### PR TITLE
Event implementation viewmodel 

### PR DESCRIPTION
**Pull request by nasvalljohan**
Calculator now behaving as a calculator should.

### CODE REVIEW

File: app/src/main/java/com/jojjnator/kalkylator/viewmodel/CalculatorViewModel.kt [hunk 1]
```diff
@@ -7,41 +7,45 @@ import com.jojjnator.kalkylator.CalculatorAction
...
             is CalculatorAction.Clear -> {
                 operator.value = ""
-                secondInput.value = 0
-                firstInput.value = 0
-                calculatedSum.value = 0
+                secondInput.value = ""
+                firstInput.value = ""
+                calculatedSum.value = 0.0
             }
             is CalculatorAction.DoCalculation -> {
```
Comment (saimanchen) [inline app/src/main/java/com/jojjnator/kalkylator/viewmodel/CalculatorViewModel.kt:28]: .DoCalculation is very repetitive, you should be able to extract this to a function or make it shorter by making an enum class

**Comment by nasvalljohan** [comment]
Review again plz

File: app/src/main/java/com/jojjnator/kalkylator/ui/Calculator.kt [hunk 2]
```diff
@@ -71,22 +96,22 @@ fun Calculator() {
         ) {
             CalculatorButton(
                 modifier = modifierSmallBox
-                    .clickable { },
+                    .clickable { viewModel.checkInput(CalculatorAction.SetNumber(7)) },
                 operator = "7"
```
Comment (saimanchen) [inline app/src/main/java/com/jojjnator/kalkylator/ui/Calculator.kt:100]: Numbers are not operators, maybe make a variable for numbers instead?

File: app/src/main/java/com/jojjnator/kalkylator/viewmodel/CalculatorViewModel.kt [hunk 3]
```diff
@@ -4,58 +4,84 @@ import androidx.compose.runtime.mutableStateOf
...
-                }
-                if (operator.value == "*") {
-                    state.value.toInt() * state2.value.toInt()
-                }
-                if (operator.value == "/") {
-                    state.value.toDouble() / state2.value.toDouble()
-                }
+                doCalculation(operatorInput = operator.value)
             }
             is CalculatorAction.Comma -> {
```
Comment (saimanchen) [inline app/src/main/java/com/jojjnator/kalkylator/viewmodel/CalculatorViewModel.kt:31]: These conditions could be one liners:

when (actionFromUI) {
    is CalculatorAction.Comma -> addCommaToDigit()
}

Comment (saimanchen) [reply app/src/main/java/com/jojjnator/kalkylator/viewmodel/CalculatorViewModel.kt:31]: Otherwise, I think, it looks great

**Review by saimanchen — COMMENTED** [review]
Looks great, but left a few comments!

**Review by saimanchen — APPROVED** [review]
Forgot to select Approve

**Pull request merged**